internal/venice-common/src/main/resources/avro/MetadataResponseRecord/v1/MetadataResponseRecord.avsc

```diff
@@ -0,0 +1,117 @@
+{
+  "type": "record",
+  "name": "MetadataResponseRecord",
+  "namespace": "com.linkedin.venice.metadata.response",
+  "doc": "This record will store version properties, key & value schemas, and routing information",
+  "fields": [
+    {
+      "name": "version_metadata",
+      "doc": "The current version number and other version properties such as the compression strategy",
+      "type": [
+        "null",
+        {
+          "name": "VersionProperties",
+          "type": "record",
+          "fields": [
+            {
+              "name": "current_version",
+              "doc": "Current version number",
+              "type": "int"
+            },
+            {
+              "name": "versions",
+              "doc": "List of all version numbers",
+              "type": {
+                "type": "array",
+                "items": "int"
+              }
+            },
+            {
+              "name": "compression_strategy",
+              "doc": "The current version's compression strategy",
+              "type": {
+                "name": "CompressionStrategy",
+                "type": "enum",
+                "symbols": [
+                  "NO_OP",
+                  "GZIP",
+                  "ZSTD",
+                  "ZSTD_WITH_DICT"
```

Comment:
Yes, will add the dictionary as another field in VersionProperties

Reply:
Ok, but let's think it through:

1. The dictionary is immutable per version so it could be a bit bloated to ship it across for each request. If the server is smart enough to send only the diff back maybe that's fine, but it's maybe too fancy. The schemas fall in the same category BTW (immutable).
2. The dictionary contains fragments of the data so we want to protect with the same level of caution as the data itself. The endpoint that serves it should have the same security controls as those which serve the data. If we intend to make this new endpoint world readable, then let's exclude the dictionary from it.